File: macOs/Sources/DASCRM/Views/Automations/AutomationsView.swift
"""
AutomationsView.swift — DAS CRM macOS
Workflow Automation Engine with Rules and Triggers
Feature parity with Android AutomationsScreen.tsx
"""

import SwiftUI

struct AutomationRule: Identifiable {
    let id: String
    let name: String
    let description: String
    let trigger: String
    let action: String
    let isActive: Bool
    let executionCount: Int
    let lastExecuted: String
}

let automationTriggers = [
    "LEAD_CREATED",
    "DEAL_UPDATED",
    "EMAIL_OPENED",
    "CALL_COMPLETED",
    "MEETING_SCHEDULED",
    "QUOTATION_SENT",
]

let automationActions = [
    "SEND_EMAIL",
    "ASSIGN_TASK",
    "UPDATE_FIELD",
    "NOTIFY_TEAM",
    "CREATE_ACTIVITY",
    "SEND_SMS",
]

let fallbackAutomations = [
    AutomationRule(id: "a1", name: "Auto-assign new leads", description: "Automatically assign leads to sales reps",
                  trigger: "LEAD_CREATED", action: "ASSIGN_TASK", isActive: true, executionCount: 342, lastExecuted: "2 hours ago"),
    AutomationRule(id: "a2", name: "Send follow-up email", description: "Send email 24h after lead creation",
                  trigger: "LEAD_CREATED", action: "SEND_EMAIL", isActive: true, executionCount: 156, lastExecuted: "1 day ago"),
    AutomationRule(id: "a3", name: "Update deal stage", description: "Auto-update deal status based on activity",
                  trigger: "CALL_COMPLETED", action: "UPDATE_FIELD", isActive: true, executionCount: 89, lastExecuted: "3 days ago"),
    AutomationRule(id: "a4", name: "Meeting notification", description: "Notify team when meeting is scheduled",
                  trigger: "MEETING_SCHEDULED", action: "NOTIFY_TEAM", isActive: false, executionCount: 0, lastExecuted: "Never"),
    AutomationRule(id: "a5", name: "Quotation reminder", description: "Send reminder 3 days before quotation expires",
                  trigger: "QUOTATION_SENT", action: "SEND_EMAIL", isActive: true, executionCount: 23, lastExecuted: "5 days ago"),
]

@MainActor
class AutomationsViewModel: ObservableObject {
    @Published var automations: [AutomationRule] = fallbackAutomations
    @Published var search: String = ""

    var filteredAutomations: [AutomationRule] {
        automations.filter { automation in
            if search.isEmpty {
                return true
            } else {
                let q = search.lowercased()
                return automation.name.lowercased().contains(q) ||
                       automation.trigger.lowercased().contains(q) ||
                       automation.action.lowercased().contains(q)
            }
        }
    }
}

struct AutomationsView: View {
    @StateObject private var viewModel = AutomationsViewModel()
    @State private var selectedAutomation: AutomationRule?
    @State private var showAutomationDetails = false
    @State private var showCreateAutomation = false

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Header
            VStack(alignment: .leading, spacing: 12) {
                Text("⚙️ Workflow Automations")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(.white)

                // Search input
                HStack(spacing: 8) {
                    Image(systemName: "magnifyingglass")
                        .foregroundColor(Color(red: 0.40, green: 0.45, blue: 0.52))
                    TextField("🔍 Search by rule name, trigger...", text: $viewModel.search)
                        .textFieldStyle(.plain)
                        .foregroundColor(.white)

                    if !viewModel.search.isEmpty {
                        Button(action: { viewModel.search = "" }) {
                            Image(systemName: "xmark.circle.fill")
                                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                        }
                        .buttonStyle(.plain)
                    }
                }
                .padding(.horizontal, 12)
                .padding(.vertical, 8)
                .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                .cornerRadius(8)

                // Action buttons
                HStack {
                    Button(action: { showCreateAutomation = true }) {
                        Text("➕ New Automation")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 12)
                            .padding(.vertical, 6)
                            .background(Color(red: 0.06, green: 0.68, blue: 0.50))
                            .cornerRadius(6)
                    }
                    .buttonStyle(.plain)

                    Spacer()
                }
            }
            .padding(16)
            .background(Color(red: 0.06, green: 0.09, blue: 0.16))
            .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)

            // Automations Table
            Table(viewModel.filteredAutomations) {
                TableColumn("Rule Name", value: \.name)
                TableColumn("Trigger", value: \.trigger)
                TableColumn("Action", value: \.action)
                TableColumn("Status") { automation in
                    Text(automation.isActive ? "ACTIVE" : "INACTIVE")
                        .foregroundColor(automation.isActive ? Color(red: 0.2, green: 0.83, blue: 0.60) : Color(red: 0.98, green: 0.30, blue: 0.40))
                }
                TableColumn("Executions") { automation in
                    Text(String(automation.executionCount))
                        .foregroundColor(Color(red: 0.38, green: 0.65, blue: 0.98))
                }
                TableColumn("Last Run", value: \.lastExecuted)
                TableColumn("Action") { automation in
                    Button(action: {
                        selectedAutomation = automation
                        showAutomationDetails = true
                    }) {
                        Text("👁️ View")
                            .font(.system(size: 9, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 8)
                            .padding(.vertical, 4)
                            .background(Color(red: 0.31, green: 0.27, blue: 0.90))
                            .cornerRadius(4)
                    }
                    .buttonStyle(.plain)
                }
            }
            .background(Color(red: 0.03, green: 0.04, blue: 0.07))
            .onDoubleClickSelectAll(false)
        }
        .background(Color(red: 0.06, green: 0.08, blue: 0.14))
        .navigationTitle("Automations")
        .sheet(isPresented: $showAutomationDetails) {
            if let automation = selectedAutomation {
                AutomationDetailsSheet(automation: automation, isPresented: $showAutomationDetails)
            }
        }
        .sheet(isPresented: $showCreateAutomation) {
            CreateAutomationSheet(automations: $viewModel.automations, isPresented: $showCreateAutomation)
        }
    }
}

struct AutomationDetailsSheet: View {
    let automation: AutomationRule
    @Binding var isPresented: Bool
    @State private var showEditAlert = false
    @State private var showToggleAlert = false

    var body: some View {
        NavigationView {
            ScrollView {
                VStack(alignment: .leading, spacing: 12) {
                    // Header
                    HStack {
                        VStack(alignment: .leading, spacing: 4) {
                            Text("⚙️ \(automation.name)")
                                .font(.system(size: 16, weight: .bold))
                                .foregroundColor(.white)
                        }
                        Spacer()

                        Text(automation.isActive ? "ACTIVE" : "INACTIVE")
                            .font(.system(size: 9, weight: .bold))
                            .foregroundColor(automation.isActive ? Color(red: 0.2, green: 0.83, blue: 0.60) : Color(red: 0.98, green: 0.30, blue: 0.40))
                            .padding(.horizontal, 8)
                            .padding(.vertical, 4)
                            .background(
                                (automation.isActive ? Color(red: 0.2, green: 0.83, blue: 0.60) : Color(red: 0.98, green: 0.30, blue: 0.40)).opacity(0.15)
                            )
                            .cornerRadius(4)
                    }
                    .padding(12)
                    .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                    .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                    .cornerRadius(12)

                    // Automation Info
                    VStack(alignment: .leading, spacing: 8) {
                        Text("📋 Automation Details")
                            .font(.system(size: 11, weight: .bold))
                            .foregroundColor(.white)

                        DetailRow(label: "Name", value: automation.name)
                        DetailRow(label: "Description", value: automation.description)
                        DetailRow(label: "Trigger", value: automation.trigger)
                        DetailRow(label: "Action", value: automation.action)
                    }
                    .padding(12)
                    .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                    .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                    .cornerRadius(12)

                    // Statistics
                    VStack(alignment: .leading, spacing: 8) {
                        Text("📊 Statistics")
                            .font(.system(size: 11, weight: .bold))
                            .foregroundColor(.white)

                        DetailRow(label: "Executions", value: String(automation.executionCount))
                        DetailRow(label: "Last Run", value: automation.lastExecuted)
                    }
                    .padding(12)
                    .background(Color(red: 0.06, green: 0.09, blue: 0.16))
                    .border(Color(red: 0.12, green: 0.16, blue: 0.23), width: 1)
                    .cornerRadius(12)

                    Spacer()

                    // Action Buttons
                    HStack(spacing: 8) {
                        Button(action: { showEditAlert = true }) {
                            Text("✏️ Edit")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.98, green: 0.58, blue: 0.09))
                                .cornerRadius(6)
                        }
                        Button(action: { showToggleAlert = true }) {
                            Text("🔄 Toggle")
                                .font(.system(size: 10, weight: .bold))
                                .foregroundColor(.white)
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.31, green: 0.27, blue: 0.90))
                                .cornerRadius(6)
                        }
                    }
                }
                .padding(16)
            }
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))
            .navigationTitle("Automation Details")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button("Close") {
                        isPresented = false
                    }
                }
            }
            .alert("✏️ Edit Automation", isPresented: $showEditAlert) {
                Button("OK") { }
            } message: {
                Text("Editing \(automation.name)...")
            }
            .alert("🔄 Toggle Automation", isPresented: $showToggleAlert) {
                Button("OK") { }
            } message: {
                Text("Toggling \(automation.name)...")
            }
        }
    }
}

struct CreateAutomationSheet: View {
    @Binding var automations: [AutomationRule]
    @Binding var isPresented: Bool

    @State private var name = ""
    @State private var description = ""
    @State private var selectedTrigger = "LEAD_CREATED"
    @State private var selectedAction = "SEND_EMAIL"
    @State private var isActive = true

    var body: some View {
        NavigationView {
            ScrollView {
                VStack(alignment: .leading, spacing: 12) {
                    Text("➕ Create New Automation")
                        .font(.system(size: 13, weight: .bold))
                        .foregroundColor(.white)

                    Text("Set up a new workflow automation rule.")
                        .font(.system(size: 10))
                        .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))

                    VStack(alignment: .leading, spacing: 8) {
                        Text("Rule Name *")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        TextField("e.g. Auto-assign new leads", text: $name)
                            .textFieldStyle(.roundedBorder)

                        Text("Description")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        TextField("What does this automation do?", text: $description)
                            .textFieldStyle(.roundedBorder)

                        Text("When (Trigger) *")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        Picker("Trigger", selection: $selectedTrigger) {
                            ForEach(automationTriggers, id: \.self) { Text($0) }
                        }
                        .pickerStyle(.menu)

                        Text("Then (Action) *")
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                        Picker("Action", selection: $selectedAction) {
                            ForEach(automationActions, id: \.self) { Text($0) }
                        }
                        .pickerStyle(.menu)

                        Toggle("Activate immediately", isOn: $isActive)
                            .foregroundColor(.white)
                    }

                    Spacer()

                    HStack(spacing: 8) {
                        Button(action: { isPresented = false }) {
                            Text("Cancel")
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.12, green: 0.16, blue: 0.23))
                                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                                .cornerRadius(6)
                        }
                        Button(action: {
                            let newAutomation = AutomationRule(
                                id: "a-\(UUID())",
                                name: name,
                                description: description,
                                trigger: selectedTrigger,
                                action: selectedAction,
                                isActive: isActive,
                                executionCount: 0,
                                lastExecuted: "Just now"
                            )
                            automations.insert(newAutomation, at: 0)
                            isPresented = false
                        }) {
                            Text("Create Automation ✓")
                                .frame(maxWidth: .infinity)
                                .padding(8)
                                .background(Color(red: 0.06, green: 0.68, blue: 0.50))
                                .foregroundColor(.white)
                                .cornerRadius(6)
                        }
                    }
                }
                .padding(16)
            }
            .background(Color(red: 0.06, green: 0.08, blue: 0.14))
            .navigationTitle("New Automation")
            .navigationBarTitleDisplayMode(.inline)
        }
    }
}

struct DetailRow: View {
    let label: String
    let value: String

    var body: some View {
        HStack {
            Text("\(label):")
                .font(.system(size: 10, weight: .bold))
                .foregroundColor(Color(red: 0.58, green: 0.64, blue: 0.68))
                .frame(width: 100, alignment: .leading)
            Text(value)
                .font(.system(size: 10))
                .foregroundColor(Color(red: 0.79, green: 0.84, blue: 0.88))
                .lineLimit(2)
            Spacer()
        }
    }
}

#Preview {
    NavigationView {
        AutomationsView()
    }
}

File: Win/ui/views/automations_view.py
"""
AutomationsView.py — DAS CRM Windows
Workflow Automation Engine with Rules and Triggers
Feature parity with Android AutomationsScreen.tsx
"""

from PyQt6.QtWidgets import (
    QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton, QLineEdit,
    QScrollArea, QFrame, QTableWidget, QTableWidgetItem, QAbstractItemView,
    QMessageBox, QDialog, QComboBox, QCheckBox, QSpinBox
)
from PyQt6.QtCore import Qt
from PyQt6.QtGui import QFont, QBrush, QColor
from dataclasses import dataclass

# ─────────────────────────────────────────────────────────────────────────────────────
# DATA MODELS
# ─────────────────────────────────────────────────────────────────────────────────────

@dataclass
class AutomationRule:
    """Represents an automation rule"""
    id: str
    name: str
    description: str
    trigger: str  # LEAD_CREATED, DEAL_UPDATED, EMAIL_OPENED, etc.
    action: str  # SEND_EMAIL, ASSIGN_TASK, UPDATE_FIELD, NOTIFY, etc.
    isActive: bool
    executionCount: int
    lastExecuted: str

TRIGGER_OPTIONS = [
    "LEAD_CREATED",
    "DEAL_UPDATED",
    "EMAIL_OPENED",
    "CALL_COMPLETED",
    "MEETING_SCHEDULED",
    "QUOTATION_SENT",
]

ACTION_OPTIONS = [
    "SEND_EMAIL",
    "ASSIGN_TASK",
    "UPDATE_FIELD",
    "NOTIFY_TEAM",
    "CREATE_ACTIVITY",
    "SEND_SMS",
]

FALLBACK_AUTOMATIONS = [
    AutomationRule("a1", "Auto-assign new leads", "Automatically assign leads to sales reps",
                  "LEAD_CREATED", "ASSIGN_TASK", True, 342, "2 hours ago"),
    AutomationRule("a2", "Send follow-up email", "Send email 24h after lead creation",
                  "LEAD_CREATED", "SEND_EMAIL", True, 156, "1 day ago"),
    AutomationRule("a3", "Update deal stage", "Auto-update deal status based on activity",
                  "CALL_COMPLETED", "UPDATE_FIELD", True, 89, "3 days ago"),
    AutomationRule("a4", "Meeting notification", "Notify team when meeting is scheduled",
                  "MEETING_SCHEDULED", "NOTIFY_TEAM", False, 0, "Never"),
    AutomationRule("a5", "Quotation reminder", "Send reminder 3 days before quotation expires",
                  "QUOTATION_SENT", "SEND_EMAIL", True, 23, "5 days ago"),
]

# ─────────────────────────────────────────────────────────────────────────────────────
# AUTOMATION DETAILS MODAL
# ─────────────────────────────────────────────────────────────────────────────────────

class AutomationDetailsModal(QDialog):
    """Modal showing detailed automation information"""
    def __init__(self, automation: AutomationRule, parent=None):
        super().__init__(parent)
        self.setWindowTitle(f"⚙️ Automation - {automation.name}")
        self.setGeometry(100, 100, 550, 500)
        self.setStyleSheet("""
            QDialog { background-color: #0f172a; }
            QLabel { color: #f8fafc; }
            QPushButton { padding: 8px 12px; border-radius: 6px; font-weight: bold; }
            QPushButton#edit { background-color: #f97316; color: white; }
            QPushButton#toggle { background-color: #4f46e5; color: white; }
            QPushButton#close { background-color: #1e293b; color: #94a3b8; }
        """)

        layout = QVBoxLayout(self)

        # Header
        headerLayout = QHBoxLayout()
        headerLayout.setContentsMargins(16, 16, 16, 12)

        titleLabel = QLabel(f"⚙️ {automation.name}")
        titleLabel.setFont(QFont("Segoe UI", 14, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        headerLayout.addWidget(titleLabel)
        headerLayout.addStretch()

        statusLabel = QLabel("ACTIVE" if automation.isActive else "INACTIVE")
        statusLabel.setFont(QFont("Segoe UI", 9, QFont.Weight.Bold))
        statusColor = "#34d399" if automation.isActive else "#ef4444"
        statusLabel.setStyleSheet(f"""
            background-color: rgba(100, 100, 100, 0.15);
            color: {statusColor};
            padding: 4px 8px;
            border: 1px solid {statusColor};
            border-radius: 6px;
        """)
        headerLayout.addWidget(statusLabel)

        layout.addLayout(headerLayout)

        # Content
        contentLayout = QVBoxLayout()
        contentLayout.setContentsMargins(16, 0, 16, 12)
        contentLayout.setSpacing(12)

        # Automation Info
        infoCard = self._build_info_section(
            "📋 Automation Details",
            [
                ("Name", automation.name),
                ("Description", automation.description),
                ("Trigger", automation.trigger),
                ("Action", automation.action),
            ]
        )
        contentLayout.addWidget(infoCard)

        # Statistics
        statsCard = self._build_info_section(
            "📊 Statistics",
            [
                ("Execution Count", str(automation.executionCount)),
                ("Last Executed", automation.lastExecuted),
                ("Status", "Active" if automation.isActive else "Inactive"),
            ]
        )
        contentLayout.addWidget(statsCard)

        contentLayout.addStretch()

        layout.addLayout(contentLayout, 1)

        # Action Buttons
        actionLayout = QHBoxLayout()
        actionLayout.setContentsMargins(16, 0, 16, 16)
        actionLayout.setSpacing(8)

        btnEdit = QPushButton("✏️ Edit")
        btnEdit.setObjectName("edit")
        btnEdit.clicked.connect(lambda: QMessageBox.information(self, "Edit", f"Editing {automation.name}..."))
        actionLayout.addWidget(btnEdit)

        btnToggle = QPushButton("🔄 Toggle")
        btnToggle.setObjectName("toggle")
        btnToggle.clicked.connect(lambda: QMessageBox.information(self, "Toggle", f"Toggled {automation.name}..."))
        actionLayout.addWidget(btnToggle)

        actionLayout.addStretch()

        btnClose = QPushButton("Close")
        btnClose.setObjectName("close")
        btnClose.clicked.connect(self.accept)
        actionLayout.addWidget(btnClose)

        layout.addLayout(actionLayout)

    def _build_info_section(self, title: str, fields: list) -> QFrame:
        """Build an info section card"""
        card = QFrame()
        card.setStyleSheet("""
            QFrame {
                background-color: #0f172a;
                border: 1px solid #1e293b;
                border-radius: 12px;
                padding: 12px;
            }
        """)

        layout = QVBoxLayout(card)
        layout.setContentsMargins(12, 12, 12, 12)
        layout.setSpacing(8)

        titleLabel = QLabel(title)
        titleLabel.setFont(QFont("Segoe UI", 11, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        layout.addWidget(titleLabel)

        for fieldName, fieldValue in fields:
            fieldLayout = QHBoxLayout()

            nameLabel = QLabel(f"{fieldName}:")
            nameLabel.setFont(QFont("Segoe UI", 10, QFont.Weight.Bold))
            nameLabel.setStyleSheet("color: #94a3b8;")
            nameLabel.setMaximumWidth(120)

            valueLabel = QLabel(fieldValue)
            valueLabel.setFont(QFont("Segoe UI", 10))
            valueLabel.setStyleSheet("color: #cbd5e1;")
            valueLabel.setWordWrap(True)

            fieldLayout.addWidget(nameLabel)
            fieldLayout.addWidget(valueLabel, 1)
            layout.addLayout(fieldLayout)

        return card

# ─────────────────────────────────────────────────────────────────────────────────────
# CREATE AUTOMATION MODAL
# ─────────────────────────────────────────────────────────────────────────────────────

class CreateAutomationModal(QDialog):
    """Modal for creating new automation"""
    def __init__(self, parent=None):
        super().__init__(parent)
        self.setWindowTitle("➕ Create New Automation")
        self.setGeometry(100, 100, 550, 550)
        self.setStyleSheet("""
            QDialog { background-color: #0f172a; }
            QLabel { color: #cbd5e1; font-weight: bold; font-size: 10px; }
            QLineEdit, QComboBox { background-color: #020617; color: #ffffff;
                                  border: 1px solid #334155; border-radius: 6px; padding: 6px; }
            QPushButton#create { background-color: #10b981; color: white; }
            QPushButton#cancel { background-color: #1e293b; color: #94a3b8; }
        """)

        layout = QVBoxLayout(self)

        title = QLabel("➕ Create New Automation")
        title.setFont(QFont("Segoe UI", 13, QFont.Weight.Bold))
        title.setStyleSheet("color: #ffffff;")
        layout.addWidget(title)

        desc = QLabel("Set up a new workflow automation rule.")
        desc.setStyleSheet("color: #94a3b8; margin-bottom: 12px;")
        layout.addWidget(desc)

        # Form fields
        layout.addWidget(QLabel("Rule Name *"))
        self.nameInput = QLineEdit()
        self.nameInput.setPlaceholderText("e.g. Auto-assign new leads")
        layout.addWidget(self.nameInput)

        layout.addWidget(QLabel("Description"))
        self.descInput = QLineEdit()
        self.descInput.setPlaceholderText("What does this automation do?")
        layout.addWidget(self.descInput)

        layout.addWidget(QLabel("When (Trigger) *"))
        self.triggerCombo = QComboBox()
        self.triggerCombo.addItems(TRIGGER_OPTIONS)
        layout.addWidget(self.triggerCombo)

        layout.addWidget(QLabel("Then (Action) *"))
        self.actionCombo = QComboBox()
        self.actionCombo.addItems(ACTION_OPTIONS)
        layout.addWidget(self.actionCombo)

        self.activeCheckbox = QCheckBox("Activate immediately")
        self.activeCheckbox.setChecked(True)
        layout.addWidget(self.activeCheckbox)

        layout.addStretch()

        btnLayout = QHBoxLayout()
        btnCancel = QPushButton("Cancel")
        btnCancel.setObjectName("cancel")
        btnCancel.clicked.connect(self.reject)
        btnCreate = QPushButton("Create Automation ✓")
        btnCreate.setObjectName("create")
        btnCreate.clicked.connect(self.accept)
        btnLayout.addWidget(btnCancel, 1)
        btnLayout.addWidget(btnCreate, 1)
        layout.addLayout(btnLayout)

    def get_automation(self) -> AutomationRule:
        """Return created automation"""
        return AutomationRule(
            id=f"a-{id(self)}",
            name=self.nameInput.text().strip(),
            description=self.descInput.text().strip(),
            trigger=self.triggerCombo.currentText(),
            action=self.actionCombo.currentText(),
            isActive=self.activeCheckbox.isChecked(),
            executionCount=0,
            lastExecuted="Never"
        )

# ─────────────────────────────────────────────────────────────────────────────────────
# MAIN AUTOMATIONS VIEW
# ─────────────────────────────────────────────────────────────────────────────────────

class AutomationsView(QWidget):
    """Workflow Automation Engine"""

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setStyleSheet("""
            QWidget { background-color: #090d16; }
            QLabel { color: #f8fafc; }
            QPushButton { color: white; font-weight: bold; border: none; border-radius: 6px; }
            QLineEdit { background-color: #0f172a; color: #ffffff; border: 1px solid #1e293b;
                       border-radius: 6px; padding: 8px; }
        """)

        self.automationsList = list(FALLBACK_AUTOMATIONS)
        self.search = ""

        self._build_ui()

    def _build_ui(self):
        """Build automations UI"""
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)

        scrollArea = QScrollArea()
        scrollArea.setWidgetResizable(True)
        scrollArea.setStyleSheet("QScrollArea { border: none; background-color: #090d16; }")

        scrollWidget = QWidget()
        scrollLayout = QVBoxLayout(scrollWidget)
        scrollLayout.setContentsMargins(16, 16, 16, 24)
        scrollLayout.setSpacing(12)

        # Title
        titleLabel = QLabel("⚙️ Workflow Automations")
        titleLabel.setFont(QFont("Segoe UI", 14, QFont.Weight.Bold))
        titleLabel.setStyleSheet("color: #ffffff;")
        scrollLayout.addWidget(titleLabel)

        # Search & Filter Bar
        searchLayout = QVBoxLayout()
        searchLayout.setContentsMargins(0, 0, 0, 0)
        searchLayout.setSpacing(8)

        # Search input
        self.searchInput = QLineEdit()
        self.searchInput.setPlaceholderText("🔍 Search by rule name, trigger...")
        self.searchInput.setMinimumHeight(32)
        self.searchInput.textChanged.connect(self._on_search_changed)
        searchLayout.addWidget(self.searchInput)

        # Action buttons
        actionLayout = QHBoxLayout()
        btnAdd = QPushButton("➕ New Automation")
        btnAdd.setStyleSheet("background-color: #10b981; padding: 6px 12px;")
        btnAdd.clicked.connect(self._open_create_automation)
        actionLayout.addWidget(btnAdd)

        actionLayout.addStretch()
        searchLayout.addLayout(actionLayout)

        scrollLayout.addLayout(searchLayout)

        # Automations Table
        self.automationsTable = QTableWidget()
        self.automationsTable.setColumnCount(6)
        self.automationsTable.setHorizontalHeaderLabels([
            "Rule Name", "Trigger", "Action", "Status", "Executions", "Last Run"
        ])
        self.automationsTable.horizontalHeader().setStretchLastSection(False)
        self.automationsTable.setSelectionBehavior(QAbstractItemView.SelectionBehavior.SelectRows)
        self.automationsTable.setSelectionMode(QAbstractItemView.SelectionMode.SingleSelection)
        self.automationsTable.setStyleSheet("""
            QTableWidget { background-color: #030712; gridline-color: #1e293b; }
            QHeaderView::section { background-color: #0b1329; color: #818cf8; padding: 6px;
                                   border: none; border-right: 1px solid #1e293b; }
        """)

        self.automationsTable.setColumnWidth(0, 180)
        self.automationsTable.setColumnWidth(1, 140)
        self.automationsTable.setColumnWidth(2, 130)
        self.automationsTable.setColumnWidth(3, 90)
        self.automationsTable.setColumnWidth(4, 100)
        self.automationsTable.setColumnWidth(5, 120)

        self.automationsTable.doubleClicked.connect(self._open_automation_details)

        self._refresh_automations_table()

        scrollLayout.addWidget(self.automationsTable, 1)

        scrollArea.setWidget(scrollWidget)
        layout.addWidget(scrollArea)

    def _refresh_automations_table(self):
        """Refresh automations table"""
        filtered = self._get_filtered_automations()

        self.automationsTable.setRowCount(len(filtered))

        for rowIdx, automation in enumerate(filtered):
            self.automationsTable.setItem(rowIdx, 0, QTableWidgetItem(automation.name))
            self.automationsTable.setItem(rowIdx, 1, QTableWidgetItem(automation.trigger))
            self.automationsTable.setItem(rowIdx, 2, QTableWidgetItem(automation.action))

            statusItem = QTableWidgetItem("ACTIVE" if automation.isActive else "INACTIVE")
            statusColor = "#34d399" if automation.isActive else "#ef4444"
            statusItem.setForeground(QBrush(QColor(statusColor)))
            self.automationsTable.setItem(rowIdx, 3, statusItem)

            countItem = QTableWidgetItem(str(automation.executionCount))
            countItem.setForeground(QBrush(QColor("#60a5fa")))
            self.automationsTable.setItem(rowIdx, 4, countItem)

            self.automationsTable.setItem(rowIdx, 5, QTableWidgetItem(automation.lastExecuted))

    def _get_filtered_automations(self) -> list:
        """Get filtered automations"""
        result = []

        for automation in self.automationsList:
            # Search filter
            if self.search.strip():
                q = self.search.lower()
                matches = (
                    q in automation.name.lower() or
                    q in automation.trigger.lower() or
                    q in automation.action.lower()
                )
                if not matches:
                    continue

            result.append(automation)

        return result

    def _on_search_changed(self):
        """Handle search input changed"""
        self.search = self.searchInput.text()
        self._refresh_automations_table()

    def _open_create_automation(self):
        """Open create automation modal"""
        dialog = CreateAutomationModal(self)
        if dialog.exec() == QDialog.DialogCode.Accepted:
            automation = dialog.get_automation()
            self.automationsList.insert(0, automation)
            self._refresh_automations_table()
            QMessageBox.information(self, "✓ Automation Created", f"Created automation: {automation.name}")

    def _open_automation_details(self, index):
        """Open automation details modal"""
        row = index.row()
        filtered = self._get_filtered_automations()

        if row < len(filtered):
            automation = filtered[row]
            dialog = AutomationDetailsModal(automation, self)
            dialog.exec()
